sql/plan/insert.go

```diff
@@ -245,28 +271,35 @@ func (i *insertIter) Next() (returnRow sql.Row, returnErr error) {
 	}
 
 	if i.replacer != nil {
-		toReturn := row.Append(row)
-		if err = i.replacer.Delete(i.ctx, row); err != nil {
-			if !sql.ErrDeleteRowNotFound.Is(err) {
-				_ = i.rowSource.Close(i.ctx)
-				return nil, err
-			}
-			// if the row was not found during deletion, write nils into the toReturn row
-			for i := range row {
-				toReturn[i] = nil
-			}
+		toReturn := make(sql.Row, len(row)*2)
+		for i := 0; i < len(row); i++ {
+			toReturn[i+len(row)] = row[i]
 		}
-
-		if err = i.replacer.Insert(i.ctx, row); err != nil {
-			_ = i.rowSource.Close(i.ctx)
-			return nil, err
+		// May have multiple duplicate pk & unique errors due to multiple indexes
+		for {
+			if err := i.replacer.Insert(i.ctx, row); err != nil {
+				if !sql.ErrPrimaryKeyViolation.Is(err) && !sql.ErrUniqueKeyViolation.Is(err) {
+					return i.ignoreOrClose(err)
```

Comment:
This is not a thing. `REPLACE IGNORE` does not exist AFAIK. This should be reverted to the original logic.